hat/app/org/hatdex/hat/api/controllers/devices/DeviceVerification.scala

```diff
@@ -0,0 +1,307 @@
+package org.hatdex.hat.api.controllers.devices
+
+import dev.profunktor.auth.jwt
+import eu.timepit.refined._
+import eu.timepit.refined.auto._
+import eu.timepit.refined.collection.NonEmpty
+import io.dataswift.adjudicator.ShortLivedTokenOps
+import io.dataswift.adjudicator.Types.{ DeviceId, HatName, ShortLivedToken }
+import io.dataswift.models.hat.applications.Application
+import org.hatdex.hat.BearerTokenParser.BearerTokenParser
+import org.hatdex.hat.NamespaceUtils.NamespaceUtils
+import org.hatdex.hat.api.controllers.devices.Types.DeviceVerificationFailure._
+import org.hatdex.hat.api.controllers.devices.Types.DeviceVerificationSuccess._
+import org.hatdex.hat.api.controllers.devices.Types._
+import org.hatdex.hat.api.controllers.{ MachineData, RequestValidationFailure, RequestVerified }
+import org.hatdex.hat.api.service.UsersService
+import org.hatdex.hat.api.service.applications.TrustedApplicationProvider
+import org.hatdex.hat.resourceManagement.HatServer
+import org.hatdex.hat.utils.{ AuthServiceRequest, AuthServiceRequestTypes }
+import play.api.libs.json.Json
+import play.api.libs.ws.WSClient
+import play.api.mvc.Headers
+import play.api.{ Configuration, Logger }
+
+import scala.concurrent.Future
+import scala.util.{ Failure, Success }
+
+import AuthServiceRequestTypes._
+
+class DeviceVerification(
+    wsClient: WSClient,
+    configuration: Configuration) {
+  implicit val ec: scala.concurrent.ExecutionContext = scala.concurrent.ExecutionContext.global
+  private val logger                                 = Logger(this.getClass)
+
+  // -- AuthService Client
+  private val authServiceAddress =
+    configuration.underlying.getString("authservice.address")
+  private val authServiceScheme =
+    configuration.underlying.getString("authservice.scheme")
+  private val authServiceEndpoint =
+    s"${authServiceScheme}${authServiceAddress}"
+  private val authServiceSharedSecret =
+    configuration.underlying.getString("authservice.sharedSecret")
+  private val authServiceClient = new AuthServiceRequest(
```

Comment:
I agree with the variable name and not with the class name, but they should be consistent...
The client should live in the appropriate package, it should be provided as a constructor argument to this class and should only be created once in a Guice module so we do not see these repeated access to the configuration parameters it needs.

Reply:
and of course it should extend an abstract trait which would be the abstraction used here (in `DeviceVerification` which really should be named `DeviceVerficationWsClient` and also have an abstract trait and so on...)